src/sentry/relay/config/metric_extraction.py

```diff
@@ -398,49 +398,53 @@ def _convert_snuba_query_to_metrics(
 
 def convert_widget_query_to_metric(
     project: Project, widget_query: DashboardWidgetQuery, prefilling: bool
-) -> Sequence[HashedMetricSpec]:
+) -> list[HashedMetricSpec]:
     """
     Converts a passed metrics widget query to one or more MetricSpecs.
     Widget query can result in multiple metric specs if it selects multiple fields
     """
     metrics_specs: list[HashedMetricSpec] = []
 
     if not widget_query.aggregates:
-        return metrics_specs
+        for column in widget_query.columns or []:
```

Comment:
This should be checking that a column is a function or not before trying to pull out an aggregate, I think. A table with a function as a column should have that column act as an aggregate. 

eg. a table widget could have the following hypothetical (😄) `columns`: 
```js
"transaction","user_misery(300)","count()","p75(measurements.lcp)"
```

`transaction` here should be a `group-by` on the on-demand tag, but `"user_misery(300)","count()","p75(measurements.lcp)"` essentially act as aggregates. Similar in sql how you have `SELECT transaction, count() ... GROUP BY transaction`.